samples/tree.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Write your Calder code here
+ */
+
+const light1 = {
+    lightPosition: [10, 10, 10],
+    lightColor: [0.3, 0.3, 0.3],
+    lightIntensity: 256
+};
+const light2 = {
+    lightPosition: [700, 500, 50],
+    lightColor: [0.3, 0.3, 0.3],
+    lightIntensity: 100
+};
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Step 1: create geometry
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Add lights to the renderer
+renderer.addLight(light1);
+renderer.addLight(light2);
+
+// Setup leaf
+const leafColor = RGBColor.fromRGB(204, 255, 204);
+const workingLeafSphere = Shape.sphere(leafColor);
+const leafSphere = workingLeafSphere.bake();
+
+// Setup branch
+const branchColor = RGBColor.fromRGB(102, 76.5, 76.5);
+const workingBranchShape = Shape.cylinder(branchColor);
+const branchShape = workingBranchShape.bake();
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Step 2: create armature
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+const bone = Armature.define((root) => {
+    root.createPoint('base', { x: 0, y: 0, z: 0 });
+    root.createPoint('mid', { x: 0, y: 0.5, z: 0 });
+    root.createPoint('tip', { x: 0, y: 1, z: 0 });
+});
+
+const treeGen = Armature.generator();
+const tree = treeGen
+    .define('branch', 1, (root) => {
```

Comment:
I like this pattern a lot as a programmer, but as an artist (with no prior programming knowledge) it might be a little confusing with the lambdas (nested code inside it, syntax), and chaining.